### PR TITLE
VISION, MISC: add some helper vision tools and misc

### PR DESCRIPTION
**Pull request by kev-the-dev**
* add milfmt alias, which will run flake8 on your mil_common repo so you can see pep8 violations before submitting a PR
* add a cute TQDM progressbar to fix_bar.py, so the user can get a rough idea on how long the program will run instead of just idling for minutes without feedback
* Add cv_tools.py, for OpenCV helper functions/classes
  * Added threshold class, which will load color threshold values in any valid colorspace from a ros param so individual perception nodes don't have to implement this
  * Added auto_canny, an adaptive Canny threshold to reduce time spent tunny Canny threshold values
* Added EllipseFinder, a class which extends RectFinder to also allow ellipse contours

### CODE REVIEW

**Comment by mattlangford** [comment]
Do these PR comments accurately reflect the contents of this PR? It seems like there's a lot more going on

**Comment by kev-the-dev** [comment]
one sec git probs

**Comment by kev-the-dev** [comment]
Fixed.

File: perception/mil_vision/mil_vision_tools/threshold.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                        used to determine conversion_code if conversion_code is None
+        @param thresh_space All caps string that's a valid OpenCV colorspace, used to convert
+                            image before thresholding. Used to determine conversion_code if it's None
+
+        ex:
+        Threshold((0, 0, 0), (255, 255, 255)) # Thresholds BGR image in BGR colorpace
+        Threshold([25, 180 180], np.array([50, 190, 200]), thresh_space='LAB')  # Thresholds BGR image in LAB
+        Threshold((50, 50, 50), [200, 200, 200], in_space='HSV', thresh_space='LAB') # Threshold HSV image in LAB
+        '''
+        assert isinstance(low, (tuple, list, np.ndarray)), 'param lower must be a tuple/list'
```
Comment (DSsoto) [inline perception/mil_vision/mil_vision_tools/threshold.py]: Add `/np.ndarray` to the error message

**Review by DSsoto — APPROVED** [review]
Looks like a pretty convenient class.
Great Documentation!

**Comment by kev-the-dev** [comment]
ellipse_finder

**Pull request merged**